### PR TITLE
feat(edit-content): Show restrictions based on the relationship.

### LINKED ISSUE #30928: Relationship Field: Allow Choosing Content According to the Relationship
### Parent Issue

https://github.com/dotCMS/core/issues/30495

## User Story

As a user, I want to select content based on its relationship to other items, so that I can establish meaningful connections between content pieces without confusion.

Screenshot 2024-12-11 at 3 53 18 PM

### 1. Button Behavior Based on Relationship Type

- **For 1-to-1 relationships:**
  - The "Existing Content" button should be disabled if the user has selected exactly one content item.
  - For 1-1  relationships, should the table allow the selection of a single content item using a radio button

- **For 1-to-n or n-to-n relationships:**
  - The "Relate" button should always be enabled, allowing multiple selections.
  - Ensure proper validation and handling for multiple content selections.

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

#30928 

### Proposed Changes

This pull request includes several changes to improve the functionality and maintainability of the `dot-select-existing-content` component and its associated files. The most important changes include enhancements to the selection mode, updates to the dialog behavior, and improvements to the component's initialization and testing.

### Enhancements to selection mode:

* Added support for dynamic selection mode in the `dot-select-existing-content.component.html` file, allowing for both multiple and single selection modes. [[1]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011R4-R7) [[2]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011L41-R24) [[3]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011L54-R37) [[4]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011R66-R71) [[5]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011R84-R88) [[6]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011L117-R121)

### Updates to dialog behavior:

* Modified the `dot-select-existing-content.component.ts` to include the `DynamicDialogConfig` and `DynamicDialogRef` services for better dialog management. [[1]](diffhunk://#diff-14d3d666b0a074fe26436013ebf4abe96e518e28e9799b85b393d14b82ac528bL1-R13) [[2]](diffhunk://#diff-14d3d666b0a074fe26436013ebf4abe96e518e28e9799b85b393d14b82ac528bL28-R35) [[3]](diffhunk://#diff-14d3d666b0a074fe26436013ebf4abe96e518e28e9799b85b393d14b82ac528bL43) [[4]](diffhunk://#diff-14d3d666b0a074fe26436013ebf4abe96e518e28e9799b85b393d14b82ac528bL54-R58)
* Updated the `dot-select-existing-content.component.spec.ts` to include tests for dialog initialization, closing behavior, and selection mode validation. [[1]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767L4-R4) [[2]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R18) [[3]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R34-R58) [[4]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767L58-R143) [[5]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R163-R207)

### Improvements to initialization and testing:

* Added initialization logic in the `dot-select-existing-content.component.ts` to ensure the component starts with the correct configuration.
* Enhanced the test suite in `dot-select-existing-content.component.spec.ts` to cover various scenarios, including initialization, dialog behavior, and item selection. [[1]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767L4-R4) [[2]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R18) [[3]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R34-R58) [[4]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767L58-R143) [[5]](diffhunk://#diff-c67ee6b7f401324d58934e1bbad99e6b1d0c2e462d5471caa5a9df58eac71767R163-R207)

### Minor adjustments:

* Updated the SCSS file `_radiobutton.scss` to include a new class for styling radio buttons.
* Added readonly properties for injected services in the `dot-select-existing-file.component.ts` to improve code readability and maintainability. [[1]](diffhunk://#diff-18bf6e06cc08560259db33d84f324194172c43e47220f07b1de0990cf71077acR46-R49) [[2]](diffhunk://#diff-18bf6e06cc08560259db33d84f324194172c43e47220f07b1de0990cf71077acR66-R69)

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

This PR fixes: #30928

### CODE REVIEW

**Comment by nicobytes** [comment]
QA Notes:

Behavior

https://github.com/user-attachments/assets/0f99677c-2cf0-424d-a073-9e3c5371397f

Dialog height

Screenshot 2024-12-27 at 11 24 22 AM

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts [hunk 1]
```diff
@@ -50,45 +55,62 @@ import { RelationshipFieldStore } from './store/relationship-field.store';
...
+                ),
+                disabled: isDisabledCreateNewContent,
+                command: () => {
+                    this.showExistingContentDialog();
+                }
+            },
+            {
+                label: this.#dotMessageService.get('dot.file.relationship.field.table.new.content'),
+                command: () => {
+                    // TODO: Implement new content
```
Comment (rjvelazco) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts:109]: Will this be in another ticket?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts [hunk 2]
```diff
@@ -98,23 +120,41 @@ export class DotEditContentRelationshipFieldComponent implements ControlValueAcc
...
+
+                if (cardinality === null) {
+                    return;
+                }
+
+                this.store.setCardinality(cardinality);
+            },
+            {
+                allowSignalWrites: true
+            }
```
Comment (rjvelazco) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts:143]: I thought this was enabled by default

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html [hunk 3]
```diff
@@ -54,7 +56,10 @@
             </td>
             <td style="min-width: 8rem">{{ item.language }}</td>
             <td style="min-width: 8rem">
+                {{ item.modDate }}
+                <!--
                 <p-chip [styleClass]="item.state.styleClass" [label]="item.state.label" />
+                -->
```
Comment (rjvelazco) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html:62]: Will this `chip` be used?

**Review by rjvelazco — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts (hunk 1 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts:109]: Yeah, this command will tackle in another card.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts (hunk 2 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.ts:143]: Is correct in v19 this option is not necessary, but now with v18 we need this flag.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html (hunk 3 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html:62]: Yes, in another card we're going to use the chip component.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/models/relationship.models.ts [hunk 4]
```diff
@@ -6,3 +6,14 @@ export interface RelationshipFieldItem extends MandatoryFields {
...
 }
+
+export enum RelationshipTypes {
+    ONE_TO_ONE = '1-1',
+    ONE_TO_MANY = '1-n',
+    MANY_TO_ONE = 'n-1',
+    MANY_TO_MANY = 'n-n'
+}
+
+export type RelationshipType = `${RelationshipTypes}`;
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/models/relationship.models.ts:17]: Nice!

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**